projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -806,4 +806,8 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
         return null;
     }
+
+    protected getNextRowByIndex(nextIndex) {
+        return this.grid.dataRowList.filter(element => element.index === nextIndex)[0].element.nativeElement;
```

Comment:
use `QueryList.find` because it returns the _first_ element matching the condition and would therefore not traverse the whole dataRowList most of the time while filter always will